ui/js/graphs.js

```diff
@@ -720,16 +722,64 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
   $scope.loadingTestData = false;
 
   var testInputCreated = false;
-
+  var testArray = [];
+  var series = [];
   $scope.addTestData = function () {
-    var series = _.clone($scope.selectedSeries);
-    series.projectName = $scope.selectedProject.name;
-    $modalInstance.close(series);
+      $scope.selectedSeriesList = $scope.rightList
+      $scope.selectedSeriesList.forEach(function(selectedSeries, i) {
+          series[i] = _.clone(selectedSeries); 
+          series[i].projectName = selectedSeries.projectName;
+      })
+      $modalInstance.close(series);
   };
 
   $scope.cancel = function () {
     $modalInstance.dismiss('cancel');
   };
+  $scope.testList = [];
+  $scope.rightSelected = [];
+  $scope.rightList = [];
+  
+  $scope.unselectTest = function () {
+      $scope.testSelected.forEach(function(test) {
+          console.log(test);
+          test = JSON.parse(test);
+          console.log(test);
+          var selected = -1;
+          $scope.rightList.forEach(function(obj, i){
+              if(obj.name === test.name) {
+                  selected = i;
+              }
+          });
+          if (selected !== -1) {
+              if (test.platform === $scope.selectedPlatform &&
+                  test.projectName === $scope.selectedProject.name) {
+                  var temp = ($scope.rightList.splice(selected, 1))[0].name;
+                  console.log(temp);
+                  $scope.LeftList.push(temp);
```

Comment:
As discussed previously, we don't want this left/right stuff anymore. Could you do a search in your patch for any instance of left/right and change it to testList/selectedTestList respectively in the final patch?

Git grep can be useful for this (e.g. `git grep left ui/` from the root)